### PR TITLE
TST: GH30999 Add match=msg to all pytest.raises in tests/generic/methods/test_sample.py

### PR DESCRIPTION
**Pull request by moink**
This pull request partially addresses xref #30999  to remove bare `pytest.raises` by adding `match=msg`. It doesn't close that issue as I have only addressed one test module: pandas/tests/generic/methods/test_sample.py

The only other thing I did was move a few lines out of the `pytest.raises` context as they were setup lines that did not raise the error, in order to make it clear which pandas method was raising the error.

- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/generic/methods/test_sample.py [hunk 1]
```diff
@@ -66,69 +66,78 @@ def test_sample_lengths(self, obj):
...
     def test_sample_invalid_random_state(self, obj):
         # Check for error when random_state argument invalid.
-        with pytest.raises(ValueError):
-            obj.sample(random_state="astring!")
+        msg = (
+            "random_state must be an integer, array-like, a BitGenerator, a numpy "
+            "RandomState, or None"
+        )
+        with pytest.raises(ValueError, match=msg):
+            obj.sample(random_state="a_string")
```
Comment (MarcoGorelli) [inline pandas/tests/generic/methods/test_sample.py:74]: any reason why you removed the exclamation mark `!`?

**Review by MarcoGorelli — APPROVED** [review]
Nice, thanks @moink !

File: pandas/tests/generic/methods/test_sample.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/tests/generic/methods/test_sample.py:74]: I'm not saying it needs to be changed back, it's fine like this, was just wondering if it had any effect

Comment (moink) [reply pandas/tests/generic/methods/test_sample.py:74]: I guess it's not really a good reason - but I didn't want people to think that it was including punctuation that made it fail. I wanted to make it clear that what was failing was that it was any string at all.

In a couple of other places in various related PRs I changed random things like "aasdf" to "invalid_plot_kind" or whatever just to make the test clearer and this goes along with that.

Comment (MarcoGorelli) [reply pandas/tests/generic/methods/test_sample.py:74]: sure, makes sense - thanks for all your help with these tests!

**Comment by jreback** [comment]
thanks @moink

**Pull request merged**